File: src/test/java/br/com/blz/testjava/domain/model/ProductTest.java
package br.com.blz.testjava.domain.model;

import org.junit.Test;

import java.util.ArrayList;
import java.util.List;

import static org.junit.Assert.assertEquals;

public class ProductTest {

  @Test
  public void checkMarketableWithQuantityGreatterZero() {
    List<Warehouse> warehouses = new ArrayList<>();
    warehouses.add(new Warehouse("SP", 12, "ECOMMERCE"));

    Inventory inventory = new Inventory();
    inventory.setWarehouses(warehouses);

    Product product = new Product();
    product.setInventory(inventory);

    assertEquals(true, product.isMarketable());
  }

  @Test
  public void checkMarketableWithoutWarehouses() {
    List<Warehouse> warehouses = new ArrayList<>();

    Inventory inventory = new Inventory();
    inventory.setWarehouses(warehouses);

    Product product = new Product();
    product.setInventory(inventory);

    assertEquals(false, product.isMarketable());
  }

}

File: src/test/kotlin/br/com/blz/testjava/domain/model/ProductTest.kt
package br.com.blz.testjava.domain.model

import org.junit.jupiter.api.Assertions.assertFalse
import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.Test
import org.mockito.Mockito.`when`
import org.mockito.Mockito.mock

class ProductTest {

    @Test
    fun `Deve permitir saber se um produto e comercializavel`() {
        assertFalse(Product(1, "L'Oréal Professionnel", mockInventory(0)).isMarketable())
        assertTrue (Product(1, "L'Oréal Professionnel", mockInventory(1)).isMarketable())
    }

    private fun mockInventory(quantity: Int) =
        `when`(mock(Inventory::class.java).calculateTotalQuantity()).thenReturn(quantity).getMock<Inventory>()

}